packages/browser-wallet/src/background/recovery.ts

```diff
@@ -75,11 +81,18 @@ function getRecoverUrl(inputs: Omit<IdentityRecoveryRequestInput, 'timestamp' |
     return `${provider.metadata.recoveryStart}?${searchParams.toString()}`;
 }
 
-async function performRecovery({ providers, ...recoveryInputs }: Payload) {
+async function performRecovery() {
```

Comment:
Now is probably not the time, but this method should be split into multiple methods. It is quite difficult to get a good overview of what is happening inside.

Reply:
Yeah, that seems like a dangerous refactoring to do **now** :thinking: